### PR TITLE
Angular's touched state should only fire if user has visited the control

### PR DESCRIPTION
**Pull request by Blackbaud-AlexKingman**
Prior to this PR, the component was setting the touched state in the value's setter. This would show "touched" when the value was programmatically updated - which is NOT what Angular intended (see link below). Instead, the component should only show "touched" when the user visits the control.

https://angular.io/guide/forms#track-control-state-and-validity-with-ngmodel

### CODE REVIEW

File: src/app/visual/select-field/select-field-visual.component.html [hunk 1]
```diff
@@ -1,48 +1,138 @@
-<br/>
-<div id="screenshot-select-field-single-mode-wrapper" style="padding: 5px 0;">
-  <div id="screenshot-select-field-single-mode">
+<div
```
Comment (Blackbaud-AlexKingman) [inline src/app/visual/select-field/select-field-visual.component.html:1]: The vast majority of changes to this file are primarily code style. I also added a reactive form below to make manual testing easier 😄  .

File: src/app/public/modules/select-field/select-field.component.spec.ts [hunk 2]
```diff
@@ -188,11 +202,11 @@ describe('Select field component', () => {
 
       setValue([component.staticData[0]]);
       fixture.detectChanges();
-      expect(component.touched).toBe(1);
+      expect(component.touched).toBe(0);
 
       setValue([component.staticData[0]]);
       fixture.detectChanges();
-      expect(component.touched).toBe(1);
+      expect(component.touched).toBe(0);
```
Comment (Blackbaud-AlexKingman) [inline src/app/public/modules/select-field/select-field.component.spec.ts:223]: So, these values should have been 0 to begin with. The form should have never been `touched` when updating the form programmatically. I think this was just an oversight.

**Review by Blackbaud-ErikaMcVey — DISMISSED** [review]

File: src/app/public/modules/select-field/select-field.component.spec.ts (hunk 2 above)
Comment (Blackbaud-ErikaMcVey) [reply src/app/public/modules/select-field/select-field.component.spec.ts:223]: Ah ignore my comment then! I’m not even sure it posted but that was the one thing I asked about

File: src/app/public/modules/select-field/select-field.component.spec.ts [hunk 3]
```diff
@@ -221,6 +235,23 @@ describe('Select field component', () => {
...
+      expect(selectField.value.length).toEqual(3);
+
+      let tokens = getTokens();
+      expect(tokens.length).toEqual(3);
+
+      closeToken(0);
+      tokens = getTokens();
+      expect(selectField.value.length).toEqual(2);
+      expect(tokens.length).toEqual(2);
+    }));
```
Comment (Blackbaud-AlexKingman) [inline src/app/public/modules/select-field/select-field.component.spec.ts:268]: My changes to fix this issue required a 1-line removal of code. Guess what? That was enough to hit our code coverage gate 😢 . So, this test is a result of covering that last bit. It's completely unrelated to my changes.

**Review by Blackbaud-TrevorBurch — APPROVED** [review]

**Pull request merged**